kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -204,10 +207,18 @@ private MultipartUpload newMultipartUpload() throws IOException {
 
     try {
       return new MultipartUpload(s3.initiateMultipartUpload(initRequest).getUploadId());
-    } catch (AmazonClientException e) {
-      // TODO: elaborate on the exception interpretation. If this is an AmazonServiceException,
-      // there's more info to be extracted.
-      throw new IOException("Unable to initiate MultipartUpload: " + e, e);
+    } catch (AmazonServiceException e) {
+      if (e.getErrorType() == ErrorType.Client) {
+        // S3 documentation states that this error type means there is a problem with the request
+        // and that retrying this request will not result in a successful response. This includes
+        // errors such as incorrect access keys, invalid parameter values, missing parameters, etc.
+        // Therefore, the connector should propagate this exception and fail.
+        throw new ConnectException("Unable to initiate MultipartUpload", e);
+      }
+
+      throw new IOException("Unable to initiate MultipartUpload.", e);
+    } catch (SdkClientException e) {
```

Comment:
`SdkClientException` is inheriting from `AmazonClientException`. Why not keep the latter?

Reply:
1. `S3#initiateMultipartUpload` explicitly throws `SDKClientException` and `AmazonServiceException` so I thought it was better to be explicit
2. `AmazonClientException` does not have an `ErrorType` field

Reply:
You'll first catch `AmazonServiceException` as you do. 
The follow up will be `AmazonClientException ` for which you don't inspect the error type. 

You can always do `instanceof SdkClientException` in that block if you needed the error type. 
I'm just interested not to risk a gap with this change here. We can revisit exception handling here more holistically in a follow up